### DISCUSSION TITLE
How can I create a custom Bootstrap component using JavaScript?

### DISCUSSION THREAD

**Discussion by DevWebster** [twbs/bootstrap#38423] [Q&A] [answered]
How can I create a custom Bootstrap component using JavaScript?

**Comment by MissieKahsay** [comment, marked as answer]
Hi @DevWebster,

To create a custom Bootstrap component using JavaScript, you can extend one of Bootstrap's existing components or create a new component from scratch. For example, to extend Bootstrap's modal component to include a custom animation, you can use the ``` $.fn.modal  ``` function to add your own animation logic:

``` JavaScript

$.fn.extend({
  animateModal: function(animation) {
    var modal = this;
    modal.addClass('animated ' + animation);
    setTimeout(function() {
      modal.removeClass('animated ' + animation);
    }, 1000);
  }
});

// Example usage:
$('#myModal').modal('show').animateModal('bounce');

```

In this example, the `` animateModal `` function extends the `` $.fn.modal `` function to add an animateModal method that takes an animation name as an argument and adds the corresponding CSS class to the modal element. The setTimeout function is used to remove the animation class after the animation is complete.
